File: test_db_connection.py
import os
import mysql.connector
from mysql.connector import Error

def test_database_connection():
    print('🔍 Testing MySQL database connection...')
    
    try:
        # Database configuration
        config = {
            'host': os.getenv('DB_HOST', 'localhost'),
            'user': os.getenv('DB_USER', 'root'),
            'password': os.getenv('DB_PASSWORD', ''),
            'database': os.getenv('DB_NAME', 'your_database'),
            'port': int(os.getenv('DB_PORT', 3306))
        }
        
        print('🔧 Current database configuration:')
        print(f"Host: {config['host']}")
        print(f"User: {config['user']}")
        print(f"Database: {config['database']}")
        print(f"Port: {config['port']}")
        print(f"Password: {'***hidden***' if config['password'] else 'not set'}")
        print('---')
        
        # Create connection
        connection = mysql.connector.connect(**config)
        
        if connection.is_connected():
            print('✅ Database connection established successfully!')
            
            cursor = connection.cursor()
            
            # Test query
            cursor.execute('SELECT 1 + 1 AS result')
            result = cursor.fetchone()
            print(f'📊 Test query result: {result[0]}')
            
            # Database info
            cursor.execute('SELECT DATABASE() as current_db, VERSION() as version')
            db_info = cursor.fetchone()
            print(f'📋 Current database: {db_info[0]}')
            print(f'📋 MySQL version: {db_info[1]}')
            
            # Show tables
            cursor.execute('SHOW TABLES')
            tables = cursor.fetchall()
            table_names = [table[0] for table in tables]
            print(f'📂 Available tables: {table_names}')
            
    except Error as e:
        print('❌ Database connection failed:')
        print(f'Error code: {e.errno}')
        print(f'Error message: {e.msg}')
        
        # Common error solutions
        if e.errno == 2003:
            print('💡 Solution: Check if MySQL server is running')
        elif e.errno == 1045:
            print('💡 Solution: Check username/password credentials')
        elif e.errno == 1049:
            print('💡 Solution: Check if database name exists')
        elif e.errno == 2005:
            print('💡 Solution: Check host address')
            
    except Exception as e:
        print(f'❌ Unexpected error: {e}')
        
    finally:
        if 'connection' in locals() and connection.is_connected():
            cursor.close()
            connection.close()
            print('🔒 Connection closed successfully')

if __name__ == "__main__":
    test_database_connection()

File: test-db-connection.js
const mysql = require('mysql2/promise');
require('dotenv').config();

async function testDatabaseConnection() {
  console.log('🔍 Testing database connection...');
  
  try {
    // Create connection
    const connection = await mysql.createConnection({
      host: process.env.DB_HOST || 'localhost',
      user: process.env.DB_USER || 'root',
      password: process.env.DB_PASSWORD || '',
      database: process.env.DB_NAME || 'your_database',
      port: process.env.DB_PORT || 3306,
    });

    console.log('✅ Database connection established successfully!');
    
    // Test a simple query
    const [rows] = await connection.execute('SELECT 1 + 1 AS result');
    console.log('📊 Test query result:', rows[0]);
    
    // Show database info
    const [dbInfo] = await connection.execute('SELECT DATABASE() as current_db, VERSION() as version');
    console.log('📋 Database info:', dbInfo[0]);
    
    // Show tables
    const [tables] = await connection.execute('SHOW TABLES');
    console.log('📂 Available tables:', tables.map(table => Object.values(table)[0]));
    
    await connection.end();
    console.log('🔒 Connection closed successfully');
    
  } catch (error) {
    console.error('❌ Database connection failed:');
    console.error('Error code:', error.code);
    console.error('Error message:', error.message);
    console.error('Error details:', error.sqlMessage || error.toString());
    
    // Common error solutions
    if (error.code === 'ECONNREFUSED') {
      console.log('💡 Solution: Check if MySQL server is running');
    } else if (error.code === 'ER_ACCESS_DENIED_ERROR') {
      console.log('💡 Solution: Check username/password credentials');
    } else if (error.code === 'ER_BAD_DB_ERROR') {
      console.log('💡 Solution: Check if database name exists');
    } else if (error.code === 'ENOTFOUND') {
      console.log('💡 Solution: Check host address');
    }
  }
}

// Show current environment variables
console.log('🔧 Current database configuration:');
console.log('Host:', process.env.DB_HOST || 'localhost');
console.log('User:', process.env.DB_USER || 'root');
console.log('Database:', process.env.DB_NAME || 'your_database');
console.log('Port:', process.env.DB_PORT || 3306);
console.log('Password:', process.env.DB_PASSWORD ? '***hidden***' : 'not set');
console.log('---');

testDatabaseConnection();